homeassistant/components/bsblan/water_heater.py

```diff
@@ -0,0 +1,114 @@
+"""BSBLAN platform to control a compatible Water Heater Device."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from bsblan import BSBLANError
+
+from homeassistant.components.water_heater import (
+    STATE_ECO,
+    STATE_OFF,
+    WaterHeaterEntity,
+    WaterHeaterEntityFeature,
+)
+from homeassistant.const import ATTR_TEMPERATURE, STATE_ON, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import BSBLanConfigEntry, BSBLanData
+from .const import DOMAIN
+from .entity import BSBLanEntity
+
+PARALLEL_UPDATES = 1
+
+# Mapping between BSBLan and HA operation modes
+OPERATION_MODES = {
+    "Eco": STATE_ECO,  # Energy saving mode
+    "Off": STATE_OFF,  # Protection mode
+    "On": STATE_ON,  # Continuous comfort mode
+}
+
+OPERATION_MODES_REVERSE = {v: k for k, v in OPERATION_MODES.items()}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: BSBLanConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up BSBLAN water heater based on a config entry."""
+    data = entry.runtime_data
+    async_add_entities([BSBLANWaterHeater(data)])
+
+
+class BSBLANWaterHeater(BSBLanEntity, WaterHeaterEntity):
+    """Defines a BSBLAN water heater entity."""
+
+    _attr_name = None
+    _attr_supported_features = (
+        WaterHeaterEntityFeature.TARGET_TEMPERATURE
+        | WaterHeaterEntityFeature.OPERATION_MODE
+    )
+
+    def __init__(self, data: BSBLanData) -> None:
+        """Initialize BSBLAN water heater."""
+        super().__init__(data.coordinator, data)
+        self._attr_unique_id = f"{format_mac(data.device.MAC)}"
+        self._attr_operation_list = list(OPERATION_MODES_REVERSE.keys())
+
+        # Set temperature limits based on device capabilities
+        if data.static.min_temp.unit in ("&deg;C", "°C"):
+            self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+        else:
+            self._attr_temperature_unit = UnitOfTemperature.FAHRENHEIT
+        self._attr_min_temp = float(data.coordinator.data.dhw.reduced_setpoint.value)
+        self._attr_max_temp = float(
```

Comment:
Also low key wondering, Does the library have types? Because if this already is a `float` we can remove this here (unless you don't have a py.typed file)

Reply:
name: str = field(metadata={"alias": "name"})
    unit: str = field(metadata={"alias": "unit"})
    desc: str = field(metadata={"alias": "desc"})
    value: str = field(metadata={"alias": "value"})
    data_type: int = field(metadata={"alias": "dataType"})
    
    I think I can set it based on the data_type. Will implement this in the next version of bsblan lib. I now need to set it as float.